server/notebooks/api_views.py

```diff
@@ -31,34 +31,43 @@ def perform_destroy(self, instance):
             raise PermissionDenied
         super().perform_destroy(instance)
 
-    def perform_create(self, serializer):
-        with transaction.atomic():
-            if "owner" in self.request.data:
+    def create(self, request):
+        if "owner" in self.request.data:
+            if (
+                not self.request.data["owner"]
+                or self.request.data["owner"] == self.request.user.username
+            ):
+                owner = self.request.user
+            elif self.request.user.can_create_on_behalf_of_others:
                 User = get_user_model()
-                if (
-                    not self.request.data["owner"]
-                    or self.request.data["owner"] == self.request.user.username
-                ):
-                    owner = self.request.user
-                elif self.request.user.can_create_on_behalf_of_others:
-                    owner = get_object_or_404(User, username=self.request.data["owner"])
-                else:
-                    raise PermissionDenied
+                owner, _ = User.objects.get_or_create(username=self.request.data["owner"])
             else:
-                owner = self.request.user
+                raise PermissionDenied
+        else:
+            owner = self.request.user
 
-            if "forked_from" in self.request.data:
-                nbr_id = self.request.data["forked_from"]
+        if "forked_from" in self.request.data:
+            nbr_id = self.request.data["forked_from"]
+            try:
                 forked_from = NotebookRevision.objects.get(id=nbr_id)
-            else:
-                forked_from = None
+            except NotebookRevision.DoesNotExist:
+                raise Http404(f"Notebook with id {nbr_id} does not exist")
+        else:
+            forked_from = None
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        with transaction.atomic():
             notebook = serializer.save(owner=owner, forked_from=forked_from)
             NotebookRevision.objects.create(
                 notebook=notebook,
                 title=self.request.data["title"],
                 content=self.request.data["content"],
             )
 
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```

Comment:
I was wondering the same thing, but couldn't think of any way of doing that. Lemme know if you can think of something. Here's the source code:

https://github.com/encode/django-rest-framework/blob/89ac0a1c7ef6153312890376ac142674b28b04f1/rest_framework/mixins.py#L12

Reply:
Yeah, the code has so many redirections, it's hard to find a good place to override it. I want to say it may need to just be added to the request.data or the serializer's validated_data? I'm not sure. Will looking into this since I'm sure other have struggled with this as well.